metrics/ci_integrations/docs/features/in_progress_builds/in_progress_builds_introduction.md

```diff
@@ -0,0 +1,271 @@
+
+The `SyncCommand` should add the `--in-progress-timeout` option using the [`ArgParser.addOption`](https://pub.dev/documentation/args/latest/args/ArgParser/addOption.html) method as follows:
+
+```dart
+    argParser.addOption(
+      _inProgressTimeoutOptionName,
+      help: 'A timeout duration in minutes for in-progress builds.',
+      valueHelp: defaultInProgressTimeout.inMinutes,
+      defaultsTo: defaultInProgressTimeout.inMinutes,
+    );
+```
+
+Where the `_inProgressTimeoutOptionName` and `defaultInProgressTimeout` are constants that stores the option name and its default value respectively.
+
+Also, the `SyncCommand` should be able to parse the given `--in-progress-timeout` option value to pass it then to the `SyncConfig`. The `SyncConfig` class, in its turn, should be updated with the `inProgressTimeout` field. Consider the [Sync Algorithm](#sync-algorithm) section to know more about changes related to the `SyncConfig` class.
+
+The following class diagram demonstrates the described changes:
+
+![Sync command class diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/ci_integrations_in_progress_doc/metrics/ci_integrations/docs/features/in_progress_builds/diagrams/sync_command_class_diagram.puml)
+
+#### Sync Algorithm
+
+The algorithm synchronizes builds for the specified `source` and `destination` integrations. The in-progress builds introduction requires this algorithm to be divided into two main parts: re-sync in-progress builds and sync new builds. The second part is how the synchronization algorithm is implemented now. So the purpose of the algorithm changes is to implement the re-syncing in-progress builds.
+
+The re-syncing in-progress builds stage is described within the [Re-Sync In-Progress Builds](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/06_builds_synchronization.md#re-sync-in-progress-builds) and [Making Things Work](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/06_builds_synchronization.md#making-things-work) sections of the [Builds Synchronization](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/06_builds_synchronization.md) document. This section also covers how steps of the re-sync part should work. Thus, to implement the desired part of the sync algorithm, it is strongly recommended to follow the definitions and tips listed in the mentioned document.
+
+The following table contains methods to implement for the `CiIntegration` with a short description:
+
+|Method|Description|
+|---|---|
+|`_syncInProgressBuilds`|Re-syncs in-progress builds for the specified projects. This method represents the `Re-Sync In-Progress Builds` stage of the sync algorithm.|
+|`_syncInProgressBuild`|Re-syncs the given in-progress build and returns a new build data. If the build shouldn't be changed, returns `null`.|
```

Comment:
```suggestion
|`_fetchInProgressBuildUpdate`|Re-syncs the given in-progress build and returns a new build data. If the build shouldn't be changed, returns `null`.|
```
Would it be great to make this method name be a bit more different to avoid confusion with the `_syncInProgressBuilds`?

Reply:
Resolving this as separating algorithm stages into different classes may remove the naming problem.